frontend/src/components/BlogPostCard/BlogPostCard.tsx

```diff
@@ -0,0 +1,195 @@
+import { useEffect, useRef, useState } from "react";
+
+import { Editor, EditorRef } from "@edifice-ui/editor";
+import { Avatar, Badge, Card, Image, useDate } from "@edifice-ui/react";
+import clsx from "clsx";
+import { odeServices } from "edifice-ts-client";
+import { useTranslation } from "react-i18next";
+
+import { useActionDefinitions } from "~/features/ActionBar/useActionDefinitions";
+import { Post, PostState } from "~/models/post";
+import { useSidebarHighlightedPost } from "~/store";
+
+export type BlogPostCardProps = {
+  /**
+   * Post to display
+   */
+  post: Post;
+
+  /**
+   * Action on click
+   */
+  onClick?: (post: Post) => void;
+};
+
+export const BlogPostCard = ({ post, onClick }: BlogPostCardProps) => {
+  const { fromNow } = useDate();
+  const { t } = useTranslation();
+  const directoryService = odeServices.directory();
+  const sidebarHighlightedPost = useSidebarHighlightedPost();
+  const editorRef = useRef<EditorRef>(null);
+  const { contrib, manager, creator } = useActionDefinitions([]);
+
+  // Number of media to display on the preview card
+  const MAX_NUMBER_MEDIA_DISPLAY = 3;
+  // // Check size of screen to display the right number of media base on breakpoint
+  const [summaryContent, setSummaryContent] = useState<string>("");
+  const [mediaURLs, setMediaURLs] = useState<string[]>([]);
+
+  const getAvatarURL = (post: Post): string => {
+    return directoryService.getAvatarUrl(post.author.userId, "user");
+  };
+
+  const displayDate = (date: string) => {
+    return fromNow(date);
+  };
+
+  const handleOnClick = (post: Post) => {
+    console.log("handleOnClick", post);
+    onClick?.(post);
+  };
+
+  useEffect(() => {
+    let contentHTML = post.content;
+    if (contentHTML) {
+      const getMediaTags = /<(img|video|iframe|audio|embed)[^>]*>(<\/\1>)?/gim;
+      const getSrc = /src=(?:"|')([^"|']*)(?:"|')/;
+      const haveEmptyTags = /<([^>|/]*)\s*[^>]*>\s*\S*(&nbsp;)*<\/\1>/gim;
+      const getSummaryTags =
+        /<([h1,h2,h3,h4,h5,p,div]+)\s*[^>]*>((?!<\/\1).)*<\/\1>/gim;
+      const mediaTags = contentHTML.match(getMediaTags);
+      contentHTML = contentHTML.replace(getMediaTags, "");
+      if (mediaTags?.length) {
+        setMediaURLs(
+          mediaTags
+            .filter((tag) => tag.includes("img"))
+            .map((tag) => {
+              const srcMatch = getSrc.exec(tag);
+              return srcMatch?.length ? srcMatch[1] : "";
+            }) || [],
+        );
+      }
+
+      while (haveEmptyTags.test(contentHTML)) {
+        contentHTML = contentHTML.replace(haveEmptyTags, "");
+      }
+
+      // Check if line clamp is supported by the browser else cut the content
+      if (!CSS.supports("-webkit-line-clamp: 2")) {
+        setSummaryContent(contentHTML);
+      } else {
+        // Cut with js the 2 first tag of the content
+        const summary = contentHTML.match(getSummaryTags);
+        setSummaryContent(
+          summary && summary?.length > 2
```

Comment:
```suggestion
          summary && summary.length > 2
```
le `?` semble superflu